src/main/java/org/mockito/internal/matchers/VarargMatcher.java

```diff
@@ -9,5 +9,14 @@
 /**
  * Internal interface that informs Mockito that the matcher is intended to capture varargs.
  * This information is needed when mockito collects the arguments.
+ *
+ * @deprecated use of this interface is deprecated as the behaviour it promotes has limitations.
+ * It is not recommended for new implementations to implement this method.
+ *
+ * <p>Instead, matchers should implement the {@link org.mockito.ArgumentMatcher#type()} method.
+ * If this method returns the same raw type as a vararg parameter, then Mockito will treat the
+ * matcher as matching the entire vararg array parameter, otherwise it will be treated as matching a single element.
+ * For an example, see {@link org.mockito.ArgumentMatchers#isNull(Class)}.
  */
+@Deprecated
 public interface VarargMatcher extends Serializable {}
```

Comment:
With this new solution, I think we can remove this interface now and include it in Mockito 5

Reply:
As per [comment below](https://github.com/mockito/mockito/pull/2807#issuecomment-1362341232) I'm prefer to remove this in a separate PR, if you don't mind?

Reply:
See https://github.com/mockito/mockito/pull/2807#issuecomment-1362341232